### PR TITLE
[Fix logout needing restart of pearai] Don't persist token in class

### PR DESCRIPTION
**Pull request by andrewjhopkins**
I still need to test this but wouldn't it be better to just not persist the token in the PearAIServer class and just fetch it from the storage on each request. The logout command clears it from storage but the expression (this.getCredentials && this.pearAIAccessToken === undefined) will skip checking since this.pearAIAccessToken is not undefined after logging in. 

Closes https://github.com/trypear/pearai-app/issues/299

What changed? Feel free to be brief.

- Do not persist token on PearAIServer class. On each request call this._checkAndFetchCredentials() to get the value from storage. 

## Checklist ✅

- [ ] I have added screenshots (if UI changes are present).
- [ ] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).

### CODE REVIEW

**Comment by andrewjhopkins** [comment]
How do I get the auth callback to point to my localbuild? I see this but I don't think I am doing it correctly. 

Screenshot 2024-09-07 at 2 40 09 PM

**Comment by Himanshu-Singh-Chauhan** [comment]
@andrewjhopkins replace trypear.ai with locahost:3000, you gotta run the landing page repo for this

File: core/llm/llms/PearAIServer.ts [hunk 1]
```diff
@@ -209,42 +203,45 @@ class PearAIServer extends BaseLLM {
...
+  private async _checkAndFetchCredentials(): Promise<string | undefined> {
+    let pearAIAccessToken = undefined;
+
     try {
       let creds = undefined;
+      let pearAIRefreshToken = undefined;
 
 
-      if (this.getCredentials && this.pearAIAccessToken === undefined) {
+      if (this.getCredentials) {
```
Comment (nang-dev) [inline core/llm/llms/PearAIServer.ts:214]: Yo andrew, so theres a mistake here

what this does is that EVERY request needs to getCredentials, slowdown will be too much.


we want to do this only if this.pearAIAccessToken === undefined


Logging in or logging out should set this to undefined, which is why we need getter and setters

Comment (nang-dev) [reply core/llm/llms/PearAIServer.ts:214]: Oh i read the description. No, we dont want it to run every req.

Comment (andrewjhopkins) [reply core/llm/llms/PearAIServer.ts:214]: Gotcha. I'll close this.

**Comment by nang-dev** [comment]
oh. andrew, did u still wanna take this up?

**Comment by andrewjhopkins** [comment]
> oh. andrew, did u still wanna take this up?

If I have time this weekend I might give it another shot but someone else can take it if they want.

**Pull request closed without merging**